test/main.jl

```diff
@@ -4,30 +4,55 @@
         CompatHelper.api_hostname(::MockCI) = ""
         CompatHelper.clone_hostname(::MockCI) = ""
 
-        @test_throws ErrorException("Unknown CI Config") CompatHelper.main(ENV, MockCI())
+        @test_throws ErrorException("Unknown CI Config: MockCI") CompatHelper.main(ENV, MockCI())
     end
 
-    @testset "successful run" begin
+    @testset "successful run GitHub" begin
         mktempdir() do tmpdir
-            cd(tmpdir)
-            patches = [
-                git_clone_patch,
-                project_toml_patch,
-                clone_all_registries_patch,
-                rm_patch,
-                pr_titles_mock,
-                git_push_patch,
-                gh_pr_patch,
-                make_clone_https_patch(tmpdir),
-                decode_pkey_patch,
-                gh_get_repo_patch,
-            ]
+            cd(tmpdir) do
+                patches = [
+                    git_clone_patch,
+                    project_toml_patch,
+                    clone_all_registries_patch,
+                    rm_patch,
+                    pr_titles_mock,
+                    git_push_patch,
+                    gh_pr_patch,
+                    make_clone_https_patch(tmpdir),
+                    decode_pkey_patch,
+                    gh_get_repo_patch,
+                ]
 
-            apply(patches) do
-                withenv(
-                    "GITHUB_REPOSITORY" => "CompatHelper.jl", "GITHUB_TOKEN" => "token"
-                ) do
-                    CompatHelper.main()
+                apply(patches) do
+                    withenv(
+                        "GITHUB_REPOSITORY" => "CompatHelper.jl", "GITHUB_TOKEN" => "token"
+                    ) do
+                        CompatHelper.main()
+                    end
+                end
+            end
+        end
+    end
+
+    @testset "successful run GitLab" begin
+        mktempdir() do tmpdir
+            cd(tmpdir) do
```

Comment:
Personally, I think the suggestion below is a *bit* nicer. It doesn't indent as much and lowers the chance of over running line length.

```suggestion
        tmpdir = mktempdir()
        cd(tmpdir) do
```

Reply:
Yeah not a big deal, I don't think when it's destroyed matters much. We can leave as is and if it gets too lengthy (side-ways) we can refactor.